### PR TITLE
Upgrade the version of `transformers`

### LINKED ISSUE #7338: Security vulnerability caused by pinned versions of transformers
**Describe the bug**

https://github.com/advisories/GHSA-v68g-wm8c-6x7j

As we pinned the dependency to an earlier version of `transformers`, there are some known security vulnerabilities that cannot be easily resolve.

Can we consider unpinning the version of `transformers`? Thanks!

### PR DESCRIPTION
**Pull request by KumoLiu**
Fixes #7338

### Description

transformers' version is pinned to v4.22 since https://github.com/Project-MONAI/MONAI/issues/5157.
Updated the version refer to https://github.com/huggingface/transformers/issues/21678.

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [ ] Quick tests passed locally by running `./runtests.sh --quick --unittests  --disttests`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

**Comment by KumoLiu** [comment]
Hi @ahatamiz, Because you are the original author, I would appreciate it if you could review my updated code. Thanks!

File: requirements-dev.txt [hunk 1]
```diff
@@ -33,7 +33,7 @@ tifffile; platform_system == "Linux" or platform_system == "Darwin"
 pandas
 requests
 einops
-transformers<4.22; python_version <= '3.10'  # https://github.com/Project-MONAI/MONAI/issues/5157
+transformers>4.36.0
```
Comment (mingxin-zheng) [inline requirements-dev.txt:36]: I think 4.36.0 doesn't have the security vulns, So it can be larger or equal to `>=`

**Review by mingxin-zheng — APPROVED** [review]

**Comment by KumoLiu** [comment]
Merge this PR first, and if you run into any problems, free feel to comment.

**Pull request merged**